From: Balint Reczey <balint.reczey@canonical.com>
Date: Mon, 10 May 2021 20:46:37 +0200
Subject: test: Allow running only a subset of integration tests by setting
 SELECTED_TESTS

(cherry picked from commit fa32ced0fbe5925fdf1a7b74d6adf44565798996)
---
 test/run-integration-tests.sh | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/test/run-integration-tests.sh b/test/run-integration-tests.sh
index 036c075..47cf8a8 100755
--- a/test/run-integration-tests.sh
+++ b/test/run-integration-tests.sh
@@ -60,17 +60,19 @@ pass_deny_list() {
     return 0
 }
 
+SELECTED_TESTS="${SELECTED_TESTS:-TEST-??-*}"
+
 # Let's always do the cleaning operation first, because it destroys the image
 # cache.
 if [ $CLEAN = 1 ]; then
-    for TEST in TEST-??-* ; do
+    for TEST in $SELECTED_TESTS; do
         ( set -x ; make -C "$TEST" clean )
     done
 fi
 
 # Run actual tests (if requested)
 if [[ $args =~ [a-z] ]]; then
-    for TEST in TEST-??-* ; do
+    for TEST in $SELECTED_TESTS; do
         COUNT=$(($COUNT+1))
 
         pass_deny_list $TEST || continue
